chef/cookbooks/ceph/recipes/osd.rb

```diff
@@ -103,7 +110,15 @@
         Chef::Log.info("Ceph: Claimed #{d.name}")
         device = {}
         dev_name = d.name.gsub("/dev/", "")
-        if node["ceph"]["osd"]["journal_devices"].include?(d.name) || (node[:block_device][dev_name]["rotational"] == "0" && node["ceph"]["osd"]["use_ssd_for_journal"])
+        if node["ceph"]["osd"]["journal_devices"].include?(d.name) ||
+            (node[:block_device][dev_name]["rotational"] == "0" &&
+             node["ceph"]["osd"]["use_ssd_for_journal"] &&
+             node["ceph"]["disk_mode"] == "all")
```

Comment:
Wouldn't it be better to just turn off ``use_ssd_for_journal`` in case of disk_mode == first? Than this last check (``node["ceph"]["disk_mode"] == "all"``) should not be needed